qemu/qmp.go

```diff
@@ -722,3 +752,40 @@ func (q *QMP) ExecutePCIVFIODeviceAdd(ctx context.Context, devID, bdf, addr, bus
 	}
 	return q.executeCommand(ctx, "device_add", args, nil)
 }
+
+// ExecuteCPUDeviceAdd adds a CPU to a QEMU instance using the device_add command.
+// driver is the CPU model, cpuID must be a unique ID to identify the CPU, socketID is the socket number within
+// node/board the CPU belongs to, coreID is the core number within socket the CPU belongs to, threadID is the
+// thread number within core the CPU belongs to.
+func (q *QMP) ExecuteCPUDeviceAdd(ctx context.Context, driver, cpuID, socketID, coreID, threadID string) error {
+	args := map[string]interface{}{
+		"driver":    driver,
+		"id":        cpuID,
+		"socket-id": socketID,
+		"core-id":   coreID,
+		"thread-id": threadID,
+	}
+	return q.executeCommand(ctx, "device_add", args, nil)
+}
+
+// ExecuteQueryHotpluggableCPUs returns a slice with the list of hotpluggable CPUs
+func (q *QMP) ExecuteQueryHotpluggableCPUs(ctx context.Context) ([]HotpluggableCPU, error) {
+	response, err := q.executeCommandWithResponse(ctx, "query-hotpluggable-cpus", nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// convert response to json
```

Comment:
This is a bit weird, but I can't actually think of a better way of doing this.

However, we should wrap any returned errors.  Right now, if one of the json Marshal or UnMarshal methods failed we'd return a json error to the user, which might be a little confusing, so something like

```
if err != nil {
    return nil, fmt.Errorf("Unable to extract CPU information: %v", err)
}
```